src/main/java/org/fogbowcloud/saps/engine/utils/retry/catalog/GetTasksRetry.java

```diff
@@ -20,8 +19,9 @@ public GetTasksRetry(Catalog imageStore, ImageTaskState state, Integer limit) {
 	}
 	
 	@Override
-	public List<SapsImage> run() throws SQLException {
-		return imageStore.getIn(state, limit);
+	public List<SapsImage> run() {
+		ImageTaskState[] states = {state};
```

Comment:
why not declaring and init this variable in the constructor?

Reply:
I moved this declaration to the constructor and only use the state array in the run method